### PR TITLE
Add support for rendering turbo stream tags

### PR DESCRIPTION
**Pull request by boardfish**
### Summary

Replicates #1137 with the aim of supporting #1106.

### Other Information

I'm not sure how to get this test to pass. I've tried using `helpers.tag.turbo_stream` and `helpers.tag.template` too, but neither seems to change the output.

Desired output (whitespace formatted for readability):

```html
<turbo-stream action="update" target="area1">
    <template>
        <span>Hello, world!</span>
    </template>
</turbo-stream>
```

Actual output (whitespace formatted for readability):

```html
<span>Hello, world!</span>
<turbo-stream action="update" target="area1">
    <template>
        &lt;span&gt;Hello, world!&lt;/span&gt;
    </template>
</turbo-stream>
```

Regardless of whether `helpers.tag.turbo_stream` or `turbo_stream.update` is used, what seems to be happening is that the block that is received is concatenated to the view as normal, then done again without being marked as `html_safe` within the tags.

Implementation of the `turbo_stream` tag builder starts around [here](https://github.com/hotwired/turbo-rails/blob/6f6d8c7cfb5f18fb7b71b47c7e73691b9e7d7e2e/app/models/turbo/streams/tag_builder.rb).This is the object returned by calling `turbo_stream` against the view.

### CODE REVIEW

**Comment by boardfish** [comment]
Turbo doesn't support Rails < 6 - that seems to come right down to installation, since the builds are failing on dependencies. Was Appraisal ever added to our testing setup? I seem to remember seeing talk about it in another PR. I think that'd resolve this issue.

**Comment by Spone** [comment]
> Turbo doesn't support Rails < 6 - that seems to come right down to installation, since the builds are failing on dependencies. Was Appraisal ever added to our testing setup? I seem to remember seeing talk about it in another PR. I think that'd resolve this issue.

Yes this is still a work in progress here: https://github.com/github/view_component/pull/1225

Feel free to review :)

**Comment by Spone** [comment]
@boardfish #1225 is merged, you can rebase and add turbo as a conditional dependency!

**Comment by boardfish** [comment]
@Spone I've tried to move things over to using Appraisal, but CI seems to be failing because it doesn't recognize `turbo-rails` as having been installed. I'm not having this problem locally when I run `bundle exec appraisal install && bundle exec appraisal rake` on a version that's compatible with `turbo-rails` (Ruby 3, `rails-head`). It doesn't look to me as though the tests are running via `appraisal` in CI, but I'm not confident around the source of the issue - would you be able to look into it?

**Comment by Spone** [comment]
> It doesn't look to me as though the tests are running via `appraisal` in CI, but I'm not confident around the source of the issue - would you be able to look into it?

I guess we will have to wait for #1230 (see [this comment](https://github.com/github/view_component/pull/1230/#issuecomment-1039669413))

**Comment by Spone** [comment]
We're now waiting for #1308 :)

**Comment by reeganviljoen** [comment]
@Spone I see #1308 has been merged, are you still blocked on this ?

**Comment by joelhawksley** [comment]
@Spone @boardfish @reeganviljoen any interest in taking this on, or should I close?

**Comment by boardfish** [comment]
I'll rebase and see where that lands us – hopefully we're able to get it merged 🤞

**Comment by reeganviljoen** [comment]
@joelhawksley i had to get investge this the other day for work reasons

Tldr the tag helpers use concat wich trips up the capture

**Comment by boardfish** [comment]
Looks like we're subject to hotwired/turbo-rails#74 here. That's perhaps the final blocker – the purist in me thinks it's not worth bringing Action Cable in just to get this over the line, and that the capture compatibility patch ought to fix this. But equally I'm thinking we shouldn't be encouraging folks towards that if they can avoid it.

Also, while it's been a hot minute since I checked #1106, fixing this between ourselves and the `turbo-rails` folks would really help these two libraries to work together. They're both pretty core to the Rails experience now too, so the easier we can make it, the better.

**Comment by joelhawksley** [comment]
@boardfish I'm fine adding ActionCable to our test stack as it's present in a lot of Rails apps ❤️

**Comment by boardfish** [comment]
Awesome, I'll do that next – I expect that to get this passing. I might see if I can contribute to `turbo-rails` to get this fully resolved afterwards.

**Comment by boardfish** [comment]
Alright, all set! We now require Action Cable in the sandbox app so that the Turbo Stream tag helper works. I also found that because of the age of this PR there was still some support for Rails versions we don't test against kicking around, so I got rid of that.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ nav_order: 5
 
 ## main
 
+* Add test coverage for use of `turbo_stream` helpers in components.
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:13]:
```suggestion
* Add test coverage for use of `turbo_stream` helpers in components when `capture_compatibility_patch_enabled` is `true`.
```

**Review by joelhawksley — APPROVED** [review]
